cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -220,24 +220,24 @@ public void getMapReduceInfo(HttpRequest request, HttpResponder responder,
    * Returns status of a type specified by the type{flows,workflows,mapreduce,spark,services,schedules}.
    */
   @GET
-  @Path("/apps/{app-id}/{type}/{id}/status")
+  @Path("/apps/{app-id}/{program-type}/{program-id}/status")
   public void getStatus(HttpRequest request, HttpResponder responder,
                         @PathParam("namespace-id") String namespaceId,
                         @PathParam("app-id") String appId,
-                        @PathParam("type") String type,
-                        @PathParam("id") String id) throws Exception {
-    if (type.equals("schedules")) {
-      getScheduleStatus(responder, appId, namespaceId, id);
+                        @PathParam("program-type") String programTypeString,
```

Comment:
nit: better not to have data type in variable name. In this case, just leave it as ``type``?